src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -36,21 +36,24 @@ kafka_client_cache::kafka_client_cache(
 }
 
 client_ptr kafka_client_cache::fetch(credential_t user) {
-    auto it = _user_client_map.find(user.name);
+    auto& inner_map = _cache.get<underlying_map>();
+    auto it_map = inner_map.find(user.name);
 
     // User not found
-    if (it == _user_client_map.end()) {
+    if (it_map == inner_map.end()) {
         throw std::out_of_range(fmt::format("User {} not found", user.name));
     }
 
-    // Otherwise user found. Move it to the beginning
-    // of the frequency list. Splice will shift the items
-    // over.
-    _user_client_list.splice(
-      _user_client_list.begin(), _user_client_list, it->second);
+    // Otherwise user found. Move them to the beginning
+    // of the frequency list.
 
-    // The client is at the second of the iterator.
-    return it->second->second;
+    client_ptr client{it_map->client};
+    auto& inner_list = _cache.get<underlying_list>();
+    // Convert the map iterator to list iterator
+    auto it_list = _cache.project<underlying_list>(it_map);
+    inner_list.relocate(it_list, inner_list.begin());
```

Comment:
It think it would make sense to extract, update the timeout, and then push_front.